### PR TITLE
[SSDK-523] Remove MGLMapboxAccessToken, update documentation for MBXAccessToken

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description
- Fixes [SSDK-523](https://mapbox.atlassian.net/browse/SSDK-523)
- Removes support for legacy MGLMapboxAccessToken
- Updates documentation for modern MBXAccessToken
- The preferred approach is to provide your access token through the `MBXAccessToken` key in Info.plist
- Updates documentation for providing access token programmatically for key rotation

### Checklist
- [x] Update `CHANGELOG`

[SSDK-523]: https://mapbox.atlassian.net/browse/SSDK-523?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tatiana-yan — APPROVED** [review]
I don't know if it is worth mentioning that now it's possible to set tokens with MapboxOptions.accessToken and this way also supports token rotation. See https://github.com/mapbox/android-docs/pull/2604 for reference. It's for Android but the idea is the same.

**Comment by aokj4ck** [comment]
> I don't know if it is worth mentioning that now it's possible to set tokens with MapboxOptions.accessToken and this way also supports token rotation. See [mapbox/android-docs#2604](https://github.com/mapbox/android-docs/pull/2604) for reference. It's for Android but the idea is the same.

That's a great improvement and I'll include that. Thank you!

**Review by ahmedaly16 — APPROVED** [review]
LGTM, just a minor comment.

File: Examples/SearchExamples/MapboxMapsCategoryResultsViewController.swift [hunk 1]
```diff
@@ -3,6 +3,7 @@ import MapboxSearch
 
 class MapboxMapsCategoryResultsViewController: MapsViewController {
     let searchEngine = CategorySearchEngine()
+//    let searchEngine = CategorySearchEngine(accessToken: "<#You can pass access token manually#>")
```
[the remark this answers is not in the document]
Comment (aokj4ck) [reply Examples/SearchExamples/MapboxMapsCategoryResultsViewController.swift:6]: Yes this provides a nice hint that the API can be used with a dynamic access token for key rotation if anyone needs to use the SDK in that way 
Screenshot 2024-01-16 at 15 29 04

**Pull request merged**